pkg/osquery/runtime/osqueryinstance.go

```diff
@@ -375,19 +362,6 @@ func (i *OsqueryInstance) launch() error {
 		return i.doneCtx.Err()
 	})
 
-	// Here be dragons
-	//
-	// There are two thorny issues. First, we "invert" control of
-	// the osquery process. We don't really know when osquery will
-	// be running, so we need a bunch of retries on these connections
```

Comment:
This part _might_ still be true?

Reply:
This part didn't seem relevant to keep here anymore because we don't have the retries below this method -- the code was refactored (previously) to move those retries inside the `StartOsqueryClient` and `StartOsqueryExtensionManagerServer` methods.

Reply:
Fair... the inverted control part is true, but maybe better elsewhere.

Reply:
I'll add it back as a comment in next PR!